### PR TITLE
username and password by UID and GID from {rootPath}/etc/passwd, {rootPath}etc/group

### PR DESCRIPTION
**Pull request by kooomix**
## Describe your changes
- Utilized os/users functions to extract data (usename, groupname) from /etc/passwd, /etc/group files structure.
- Implemented MakeContaineredFileInfo and MakeHostFileInfo to get the info from container files and host files.

### CODE REVIEW

File: sensor/kubeproxysensor.go [hunk 1]
```diff
@@ -30,14 +30,11 @@ func SenseKubeProxyInfo() (*KubeProxyInfo, error) {
 	}
 
 	// kubeconfig
-	kubeConfigPath, ok := proc.GetArg(kubeConfigArgName)
+	kubeConfigPath, ok := proc.GetArg("--config")
```
Comment (dwertent) [inline sensor/kubeproxysensor.go:33]: Why did you override the const?

File: sensor/osuserswrapper.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	f, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	return findGroupId(gid, f)
+}
+
+// returns group name if gid was found in a group file {root}/etc/group, otherwise returns empty string.
+func LookupGroupnameByGID(gid int64, root string) (string, error) {
```
Comment (dwertent) [inline sensor/osuserswrapper.go:44]: Why is this a glob al function?

File: sensor/osuserswrapper.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	f, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	return findUserId(uid, f)
+}
+
+// returns username if uid was found in a users file {root}/etc/passwd, otherwise returns empty string.
+func LookupUsernameByUID(uid int64, root string) (string, error) {
```
Comment (dwertent) [inline sensor/osuserswrapper.go:67]: Why is this a glob al function?

File: sensor/utils.go [hunk 4]
```diff
@@ -103,13 +103,44 @@ func MakeFileInfo(filePath string, readContent bool) (*FileInfo, error) {
 	return &ret, nil
 }
 
-// MakeHostFileInfo is a wrapper of `MakeFileInfo` for host files
+// MakeContaineredFileInfo is a wrapper of `MakeChangedRootFileInfo` for container files
+func MakeContaineredFileInfo(filePath string, readContent bool, p *ProcessDetails) (*FileInfo, error) {
```
Comment (dwertent) [inline sensor/utils.go:107]: Why is this a glob al function?

File: sensor/utils.go [hunk 5]
```diff
@@ -103,13 +103,44 @@ func MakeFileInfo(filePath string, readContent bool) (*FileInfo, error) {
...
+	return MakeChangedRootFileInfo(filePath, readContent, p.RootDir())
+}
+
+// MakeHostFileInfo is a wrapper of `MakeChangedRootFileInfo` for host files
 func MakeHostFileInfo(filePath string, readContent bool) (*FileInfo, error) {
```
Comment (dwertent) [inline sensor/utils.go:112]: Why is this a glob al function?

File: sensor/utils.go [hunk 6]
```diff
@@ -103,13 +103,44 @@ func MakeFileInfo(filePath string, readContent bool) (*FileInfo, error) {
...
-	obj, err := MakeFileInfo(hostPath(filePath), readContent)
-	if err == nil {
-		obj.Path = filePath
+	return MakeChangedRootFileInfo(filePath, readContent, hostFileSystemDefaultLocation)
+}
+
+// MakeHostFileInfo is a wrapper of `MakeFileInfo` for rootDir/filePath
+func MakeChangedRootFileInfo(filePath string, readContent bool, rootDir string) (*FileInfo, error) {
```
Comment (dwertent) [inline sensor/utils.go:117]: Why is this a glob al function?

**Review by dwertent — CHANGES_REQUESTED** [review]

File: sensor/osuserswrapper.go (hunk 3 above)
Comment (kooomix) [reply sensor/osuserswrapper.go]: Right, fixed (will pay more attention to global/private)

File: sensor/utils.go (hunk 4 above)
Comment (kooomix) [reply sensor/utils.go]: Right, fixed

File: sensor/utils.go (hunk 6 above)
Comment (kooomix) [reply sensor/utils.go]: Right, fixed

**Pull request merged**